Instacard/models/ExtractedTextModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  ExtractableImageModel.swift
+//  Instacard
+//
+//  Created by Nick Kenyeres on 2022-12-03.
+//
+
+import Foundation
+import PhotosUI
+import Vision
+
+class ExtractedTextModel: ObservableObject {
+    let cgImage: CGImage
+    
+    enum ExtractedTextState {
+        case loading
+        case success([String])
+        case failure(Error)
+    }
+    
+    enum ExtractionError: Error {
+        case extractionFailed
+        case textNotFound
+    }
```

Comment:
I've added two different errors in case we want to distinguish between being unable to work with the image at all versus being the image having no text in it.